client/my-sites/stats/stats-period-navigation/index.jsx

```diff
@@ -198,9 +233,14 @@ class StatsPeriodNavigation extends PureComponent {
 			gateDateControl,
 			intervals,
 			siteId,
+			momentSiteZone,
 		} = this.props;
 
-		const isToday = moment( date ).isSame( moment(), period );
+		const isToday = moment( date ).isSame( momentSiteZone, period );
+
+		// TODO: Refactor the isWithNewDateFiltering dedicated variables.
+		const isChartRangeEndToday = moment( dateRange.chartEnd ).isSame( momentSiteZone, period );
```

Comment:
We should always use `momentSiteZone` for relevant date like Today, Yesterday etc. @Automattic/red